apps/api/src/purchase-orders/purchase-orders.service.ts

```diff
@@ -0,0 +1,17 @@
+import {Injectable} from '@nestjs/common';
+import {PurchaseOrders} from "@prisma/client";
+import {PrismaService} from "../prisma.service";
+
+@Injectable()
+export class PurchaseOrdersService {
+  constructor(private prisma: PrismaService) {
+  }
+
+  async findAll(): Promise<PurchaseOrders[]> {
+    return (await this.prisma.purchaseOrders.findMany({include: {purchase_order_line_items: true}})).sort((a, b) => new Date(a.expected_delivery_date).getTime() - new Date(b.expected_delivery_date).getTime());
```

Comment:
It looks like you're pulling all the purchase orders back and then sorting them in memory on the server.  What would happen if we introduced pagination as a requirement?

Reply:
Yeah, which is a very hacky approach to take and I didn't like it which is why I mentioned that in my comment below about there being a better approach. I actually just found the docs for prisma here: https://www.prisma.io/docs/orm/prisma-client/queries/crud#read

https://www.prisma.io/docs/orm/prisma-client/queries/filtering-and-sorting
Under "Sorting".

I think a better approach would be to have is sorted in our call. Something like this. 
```suggestion
    return (await this.prisma.purchaseOrders.findMany({include: {purchase_order_line_items: true}})).sort((a, b) => new Date(a.expected_delivery_date).getTime() - new Date(b.expected_delivery_date).getTime());
    
    return (await this.prisma.purchaseOrders.findMany({orderBy: [
    {
      role: 'desc',
    },
    {
      name: 'desc',
    },
  ],
  include: {purchase_order_line_items: true}
  }))
```

Reply:
Regarding the pagination question, if would only sort the first ten that we receive after the first default ten are pulled. Or whatever number we are pulling.